File: Core/Entities/Identity/UserParams.cs
namespace Core.Entities.Member.Parameters
{
    public class UserParams : PaginationParams
    {


        public string CurrentUsername { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 100;

        // Sorting functionality
        public string OrderBy { get; set; } = "lastActive";
    }
}

File: client/src/app/shared/_models/accountModels/userParams.ts
import { IUser } from './user';

export class UserParams {
  gender!: string;
  minAge = 18;
  maxAge = 100;
  pageIndex = 1;
  pageSize = 6;
  orderBy = 'lastActive';

  constructor(user: IUser) {
    this.gender = user.gender === 'female' ? 'male' : 'female';
  }
}
